src/Period.php

```diff
@@ -887,6 +887,35 @@ public function diff(self $interval): array
         ];
     }
 
+    /**
+     * Returns the difference set operation between two intervals as a Sequence.
+     * The Sequence can contain from 0 to 2 Periods depending on the result of
+     * the operation.
+     *
+     * [--------------------)
+     *          -
+     *                [-----------)
+     *          =
+     * [--------------)
+     */
+    public function substract(self $interval): Sequence
```

Comment:
why does it returns a Sequence ? Shouldn't it returns `?Period` instead 🤔

Reply:
Because A - B sometimes could result in 2 Period objects.
Example:
![image](https://user-images.githubusercontent.com/4517370/54438779-b069ae00-4737-11e9-8add-c5c0d7d6100a.png)

Reply:
Consider I used to return Period|null until I started creating the unit tests, then I realized there was a problem :)